packages/build-tools/src/steps/functions/eagerBundle.ts

```diff
@@ -0,0 +1,50 @@
+import { BuildFunction, BuildStepInput, BuildStepInputValueTypeName } from '@expo/steps';
+import semver from 'semver';
+
+import { eagerBundleAsync } from '../../common/eagerBundle';
+import { resolvePackageManager } from '../../utils/packageManager';
+
+export function eagerBundleBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'eager_bundle',
+    name: 'Eager bundle',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'resolved_eas_update_runtime_version',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    fn: async (stepsCtx, { env, inputs }) => {
+      const job = stepsCtx.global.staticContext.job;
+      if (!job.platform) {
+        throw new Error('Custom build job must have platform set');
```

Comment:
Hm, this should never happen. Or, in other words, if a user is using the function in a custom build, `job.platform` will be set, and if a user is using the function in a generic job, `job.platform` will be missing, but then "custom build job" is a wrong name.

Let's allow passing in a `platform` so people can use this in generic jobs?